src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordConstructor.cs

```diff
@@ -33,22 +33,15 @@ internal TypeDeclarationSyntax GetSyntax()
 
         protected override ParameterListSyntax GetParameterList()
         {
-            return GetSyntax() switch
-            {
-                RecordDeclarationSyntax record => record.ParameterList!,
-                RecordStructDeclarationSyntax recordStruct => recordStruct.ParameterList!,
-                _ => throw ExceptionUtilities.Unreachable
-            };
+            var record = (RecordDeclarationSyntax)GetSyntax();
+            return record.ParameterList!;
         }
 
         protected override CSharpSyntaxNode? GetInitializer()
         {
-            return GetSyntax() switch
-            {
-                RecordDeclarationSyntax record => record.PrimaryConstructorBaseType,
-                RecordStructDeclarationSyntax => null,
-                _ => throw ExceptionUtilities.Unreachable
-            };
+            var record = (RecordDeclarationSyntax)GetSyntax();
+            Debug.Assert(record.Kind() == SyntaxKind.RecordDeclaration || record.PrimaryConstructorBaseType is null);
```

Comment:
>|| record.PrimaryConstructorBaseType is null [](http://example.com/codeflow?start=71&length=44)

We cannot make this assumption. I think we should explicitly return null for a record struct.